src/components/TypingIndicator/TypingIndicator.vue

```diff
@@ -0,0 +1,153 @@
+  -
+  - @author Maksim Sukharev <antreesy.web@gmail.com>
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div v-show="typingParticipants.length" class="indicator">
+		<div class="indicator__wrapper">
+			<div class="indicator__avatars">
+				<AvatarWrapper v-for="(participant, index) in visibleParticipants"
+					:id="participant.actorId"
+					:key="index"
+					:name="participant.displayName"
+					:source="participant.actorType"
+					small
+					condensed
+					:condensed-overlap="8"
+					disable-menu
+					disable-tooltip />
+			</div>
+			<p class="indicator__main" v-html="indicatorMessage" />
+		</div>
+	</div>
+</template>
+
+<script>
+import AvatarWrapper from '../AvatarWrapper/AvatarWrapper.vue'
+
+export default {
+	name: 'TypingIndicator',
+	components: { AvatarWrapper },
+
+	props: {
+		/**
+		 * The conversation token
+		 */
+		token: {
+			type: String,
+			required: true,
+		},
+	},
+
+	computed: {
+		participants() {
+			return this.$store.getters.participantsList(this.token).filter(participant => {
+				return participant.actorId !== this.$store.getters.getActorId()
+			})
+		},
+
+		typingParticipants() {
+			// TODO implement signal receiving here
+			return this.participants
+		},
+
+		visibleParticipants() {
+			return this.typingParticipants.slice(0, 3)
+		},
+
+		hiddenParticipantsCount() {
+			return this.typingParticipants.slice(3).length
+		},
+
+		indicatorMessage() {
+			if (!this.typingParticipants) {
+				return ''
+			}
+
+			const [user1, user2, user3] = this.visibleParticipants.map(participant => this.getParticipantName(participant)).reverse()
+
+			if (this.typingParticipants.length === 1) {
+				return t('spreed', '<strong>{user1}</strong> is typing...',
+					{ user1 })
+			}
+
+			if (this.typingParticipants.length === 2) {
+				return t('spreed', '<strong>{user1}</strong> and <strong>{user2}</strong> are typing...',
+					{ user1, user2 })
+			}
+
+			if (this.typingParticipants.length === 3) {
+				return t('spreed', '<strong>{user1}</strong>, <strong>{user2}</strong> and <strong>{user3}</strong> are typing...',
+					{ user1, user2, user3 })
+			}
+
+			return n('spreed', '<strong>{user1}</strong>, <strong>{user2}</strong>, <strong>{user3}</strong> and %n other are typing...',
```

Comment:
> so it should be safe to use v-html directive

We do that already in some places, yeah.

the problem is this way enables translators to add or remove more HTML.
You can also escape the displaynames yourself and stop the escaping of parameters:
https://github.com/nextcloud/spreed/blob/d41fb8ef032f8fa4a8e2ea48c7eae5f6d60708a9/src/components/MessagesList/MessagesList.vue#L431-L438